File: c++/leetcode/693. Binary Number with Alternating Bits.cpp
class Solution
{
public:
  bool hasAlternatingBits(int n)
  {
    vector<int> digits;

    while (n)
    {
      digits.push_back(n % 2);
      n = n / 2;
    }

    for (int i = 1; i < digits.size(); i++)
    {
      if (digits[i] == digits[i - 1])
        return false;
    }

    return true;
  }
};

File: javascript/leetcode/693. Binary Number with Alternating Bits.js
/**
 * @param {number} n
 * @return {boolean}
 */
var hasAlternatingBits = function (n) {
  const digits = [];
  while (n) {
    digits.push(n % 2);
    n = parseInt(n / 2);
  }

  for (let i = 1; i < digits.length; i++) {
    if (digits[i] === digits[i - 1]) return false;
  }

  return true;
};
